spec/consensus/signing.md

```diff
@@ -30,14 +30,30 @@ All signed messages must correspond to one of these types.
 
 ## Timestamp
 
-Timestamp validation is subtle and there are currently no bounds placed on the
-timestamp included in a proposal or vote. It is expected that validators will honestly
-report their local clock time. The median of all timestamps
-included in a commit is used as the timestamp for the next block height.
+Both `Proposal` and `Vote` messages include a `Timestamp` field of
+[Time](../core/data_structures.md#time) data type.
 
+Timestamp validation is subtle and there are currently no validations on the
+timestamp included in a received `Proposal` or `Vote`. 
+As a general role, it is expected that validators report in this field their
+local clock time.
 Timestamps are expected to be strictly monotonic for a given validator, though
-this is not currently enforced.
-
+this is not enforced.
+
+Some timestamps, however, are used by the algorithms adopted for computing
+[block times](./time.md):
+
+- [BFT Time](./bft-time.md): the `Timestamp` field of `Precommit` vote messages
+  are used to compute the `Time` for the next proposed block.
+  Correct validators are expected to report their local clock time, provided
+  that the time is higher than the current block's time.
+  Otherwise, the reported time is the current block's time plus 1ms.
+
+- [PBTS](./proposer-based-timestamp/README.md): the `Timestamp` field of a
+  `Proposal` message must match the proposed `Block.Time`.
+   Otherwise, the `Proposal` will be rejected by correct validators.
+   There are no requirements for `Vote.Timestamp` values.
```

Comment:
I'm just wondering why there's not need to validate the vote timestamp, shouldn't there be some validation that the timestamp of the vote be within some bounds of the timestamp of the proposal ?

Or is that not relevant because if the validator clock is drifted too much it will consider the proposal invalid and won't sign it ? Does PBTS changes the chances of this happening, or not really because you in the timestamp validation it still relies on local clock?

Anyway, was just trying to understand if there was a possibility to have votes with drifted timestamps as part of the vote set.

Ideally, the timestamp of a vote should closely align with the timestamp of the proposal to ensure that the voting reflects a synchronized view of time across the network.

Reply:
> Or is that not relevant because if the validator clock is drifted too much it will consider the proposal invalid and won't sign it ?

Good observation. Voting for a proposal implies that the validator's clocks were synchronized within accepted bounds and therefore the timestamp field of the vote becomes useless and could even be removed at some point.

Reply:
> and could even be removed at some point

It will, for BLS